openupgrade_scripts/scripts/iap/18.0.1.1/post-migration.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2025 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        f"""
+        WITH sub AS (
+            INSERT INTO iap_service (
+                name, technical_name, integer_balance, unit_name,
+                write_uid, write_date, create_uid, create_date)
+            SELECT ia.name, ia.{openupgrade.get_legacy_name("service_name")}, TRUE,
+                jsonb_build_object('en_US', 'Credits'), ia.write_uid, ia.write_date,
+                ia.create_uid, ia.create_date
+            FROM iap_account ia
+            WHERE ia.service_id IS NULL
+            RETURNING id, ia.id as account_id
+        )
+        UPDATE iap_account ia
+        SET service_id = sub.id
+        FROM sub
+        WHERE ia.id = sub.account_id
+        """,
+    )
```

Comment:
this needs grouping by `openupgrade.get_legacy_name("service_name")`, and you should do the matching by service_name/technical_name